### PR TITLE
Changes to the names of the required headers in the `presignData` mutation response

### PR DESCRIPTION
**Pull request by AleksandrSeleznev**
### Summary

@ArthurZaharov noticed that some headers from `presignData` mutation are not properly set from a specification point of view.

### How it works

In order to upload presigned file, you have to use `Content-Disposition` and `Content-Type` headers instead of `content_disposition` and `content_type` respectively.

### Test plan

N/A

### Review notes

While reviewing pull-request (especially when it's your pull-request),
please make sure that:

- you understand what problem is solved by PR and how is it solved
- new tests are in place, no redundant tests
- DB schema changes reflect new migrations
- newly introduced DB fields have indexes and constraints
- there are no missed files (migrations, view templates)
- required ENV variables added and described in `.env.example` and added to Heroku
- associated Heroku review app works correctly with introduced changes

### Deploy notes

N/A

### References
N/A

### CODE REVIEW

File: app/interactors/prepare_presign_image.rb [hunk 1]
```diff
@@ -36,8 +36,8 @@ def location
 
   def options
     {
-      content_disposition: ContentDisposition.inline(filename),
-      content_type: type,
+      "Content-Disposition": ContentDisposition.inline(filename),
+      "Content-Type": type,
       content_length_range: 0..UPLOAD_SIZE_LIMIT
```
Comment (EvgeniyEsaulkov) [inline app/interactors/prepare_presign_image.rb:41]: Shouldn't we change `content_length_range` key to `'content-length-range'`?

**Review by EvgeniyEsaulkov — COMMENTED** [review]
We use 'POST' method to upload images to AWS for now. However we can change it to 'PUT' if it's necessary:
```ruby
def upload_params
  @upload_params ||= storage.presign(location, :put, **options).slice(:url, :headers)
end
```
But after these changes it will be impossible. Is it ok that we willn't able to use 'PUT' for direct upload in the future?

**Comment by EvgeniyEsaulkov** [comment]
@AleksandrSeleznev @ArthurZaharov I just checked `presignData` mutation on the `master` branch and it works correctly - as displayed on the screenshot below.
image

I think the issue is how these options aren't converted to the correct form for the local storage (screenshot above is for AWS S3)

File: app/interactors/prepare_presign_image.rb (hunk 1 above)
Comment (EvgeniyEsaulkov) [reply app/interactors/prepare_presign_image.rb:41]: Nevermind (see https://github.com/fs/rails-base-graphql-api/pull/143#issuecomment-812827736)

**Comment by EvgeniyEsaulkov** [comment]
aws-sdk-s3 has `PresignedPost` class that converts `:content_type` to `'Content-Type'`:
```ruby
      # @return [String]
      # @!method content_type(value)
      #   A standard MIME type describing the format of the contents.
      #   @param [String] value
      #   @see http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.21
      #   @return [self]
      #
      # @!method content_type_starts_with(prefix)
      #   @param [String] prefix
      #   @see #content_type
      #   @return [self]
      define_field(:content_type, 'Content-Type', starts_with: true)
```
https://github.com/aws/aws-sdk-ruby/blob/6b1b19e64bb672cc5809277473649f0ad884c4fb/gems/aws-sdk-s3/lib/aws-sdk-s3/presigned_post.rb#L357-L368

Probably we have to implement something similar for `LocalStorage` class.

**Comment by AleksandrSeleznev** [comment]
> aws-sdk-s3 has `PresignedPost` class that converts `:content_type` to `'Content-Type'`:
> 
> ```ruby
>       # @return [String]
>       # @!method content_type(value)
>       #   A standard MIME type describing the format of the contents.
>       #   @param [String] value
>       #   @see http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.21
>       #   @return [self]
>       #
>       # @!method content_type_starts_with(prefix)
>       #   @param [String] prefix
>       #   @see #content_type
>       #   @return [self]
>       define_field(:content_type, 'Content-Type', starts_with: true)
> ```
> 
> https://github.com/aws/aws-sdk-ruby/blob/6b1b19e64bb672cc5809277473649f0ad884c4fb/gems/aws-sdk-s3/lib/aws-sdk-s3/presigned_post.rb#L357-L368
> 
> Probably we have to implement something similar for `LocalStorage` class.

@EvgeniyEsaulkov what do you think - is it necessary to implement an additional class for this? Probably, we could create some private method inside `LocalStorage` class to prepare these options?

**Comment by AleksandrSeleznev** [comment]
> @AleksandrSeleznev @ArthurZaharov I just checked `presignData` mutation on the `master` branch and it works correctly - as displayed on the screenshot below.
> image
> 
> I think the issue is why these options aren't converted to the correct form for the local storage (screenshot above is for AWS S3)

It works fine, for sure. The main concern was that the fields are presented incorrectly. We could relate these to some cosmetic issues.

**Comment by EvgeniyEsaulkov** [comment]
> @EvgeniyEsaulkov what do you think - is it necessary to implement an additional class for this? Probably, we could create some private method inside LocalStorage class to prepare these options?

I agree - we don't need the full functionality of `PresignedPost` class. For now I suggest define some fields that we should convert and add a method that implements this conversion. Something like this:
```ruby
CONVERSION_FIELDS = {
  cache_control: 'Cache-Control',
  content_disposition: 'Content-Disposition',
  content_encoding: 'Content-Encoding',
  content_length_range: 'content-length-range',
  content_type: 'Content-Type'
}.freeze

def convert_options(options)
  CONVERSION_FIELDS.each do |key, value|
    options[value] = options.delete(key)
  end

  options.compact
end
```

**Comment by EvgeniyEsaulkov** [comment]
@AleksandrSeleznev Would you like to close this PR and open another one with the changes discussed above? Or do you want me to describe it in issue?

**Comment by AleksandrSeleznev** [comment]
> @AleksandrSeleznev Would you like to close this PR and open another one with the changes discussed above? Or do you want me to describe it in issue?

I assume that we can close this PR and open the new one in order to implement your suggestions regarding that problem.

**Comment by EvgeniyEsaulkov** [comment]
I created #161 issue for that so this PR can be closed

**Pull request closed without merging**